Move BuilderState::createFilterOperations() to a separate file

shallawa opened the pull request:
#### ea05e83e61422ff0c4a68c659f9a99aa14064121
<pre>
Move BuilderState::createFilterOperations() to a separate file
<a href="https://bugs.webkit.org/show_bug.cgi?id=246675">https://bugs.webkit.org/show_bug.cgi?id=246675</a>

Reviewed by Antti Koivisto.

Make this function a stand-alone function and make used by the CSS style builder.
The plan is to use this function the 2D canvas for the filter API.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFilterImageValue.cpp:
(WebCore::CSSFilterImageValue::createFilterOperations):
* Source/WebCore/style/ColorFromPrimitiveValue.cpp: Added.
(WebCore::colorFromPrimitiveValue):
(WebCore::colorFromPrimitiveValueWithResolvedCurrentColor):
* Source/WebCore/style/ColorFromPrimitiveValue.h: Added.
* Source/WebCore/style/FilterOperationsBuilder.cpp: Added.
(WebCore::filterOperationForType):
(WebCore::createFilterOperations):
* Source/WebCore/style/FilterOperationsBuilder.h: Added.
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertFilterOperations):
* Source/WebCore/style/StyleBuilderState.cpp:
(WebCore::Style::BuilderState::createFilterOperations):
(WebCore::Style::BuilderState::colorFromPrimitiveValue const):
(WebCore::Style::BuilderState::colorFromPrimitiveValueWithResolvedCurrentColor const):
(WebCore::Style::filterOperationForType): Deleted.
* Source/WebCore/style/StyleBuilderState.h:

Canonical link: <a href="https://commits.webkit.org/255706@main">https://commits.webkit.org/255706@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb7622f7f03e427ab7e96d7fd39172a880629bf1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93362 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2557 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/24005 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103038 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/163365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97360 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2566 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30864 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/85731 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/99144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99025 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/1798 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79815 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/28754 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/83711 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/83467 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/71817 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37247 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/17336 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35073 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18583 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3946 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38947 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41115 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40882 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37803 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

anttijk commented on Source/WebCore/style/FilterOperationsBuilder.h:
```diff
@@ -0,0 +1,32 @@
+
+#pragma once
+
+namespace WebCore {
+
```
This should go to Style namespace

anttijk commented on Source/WebCore/style/ColorFromPrimitiveValue.h:
```diff
@@ -0,0 +1,44 @@
+
+namespace Style {
+enum class ForVisitedLink : bool;
+}
```
Everything should be in Style namespace

anttijk commented on Source/WebCore/style/FilterOperationsBuilder.cpp:
```diff
@@ -0,0 +1,196 @@
+#include "Document.h"
+#include "TransformFunctions.h"
+
+namespace WebCore {
+
```
Style namespace

anttijk commented on Source/WebCore/style/ColorFromPrimitiveValue.cpp:
```diff
@@ -0,0 +1,73 @@
+#include "StyleBuilderState.h"
+#include "StyleColor.h"
+
+namespace WebCore {
```
This should be in WebCore::Style namespace

shallawa committed:
Move BuilderState::createFilterOperations() to a separate file
https://bugs.webkit.org/show_bug.cgi?id=246675

Reviewed by Antti Koivisto.

Make this function a stand-alone function and make used by the CSS style builder.
The plan is to use this function the 2D canvas for the filter API.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFilterImageValue.cpp:
(WebCore::CSSFilterImageValue::createFilterOperations):
* Source/WebCore/style/ColorFromPrimitiveValue.cpp: Added.
(WebCore::colorFromPrimitiveValue):
(WebCore::colorFromPrimitiveValueWithResolvedCurrentColor):
* Source/WebCore/style/ColorFromPrimitiveValue.h: Added.
* Source/WebCore/style/FilterOperationsBuilder.cpp: Added.
(WebCore::filterOperationForType):
(WebCore::createFilterOperations):
* Source/WebCore/style/FilterOperationsBuilder.h: Added.
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertFilterOperations):
* Source/WebCore/style/StyleBuilderState.cpp:
(WebCore::Style::BuilderState::createFilterOperations):
(WebCore::Style::BuilderState::colorFromPrimitiveValue const):
(WebCore::Style::BuilderState::colorFromPrimitiveValueWithResolvedCurrentColor const):
(WebCore::Style::filterOperationForType): Deleted.
* Source/WebCore/style/StyleBuilderState.h:

Canonical link: https://commits.webkit.org/255706@main
